### DISCUSSION TITLE
MPD Raspi works on Buster but not Bullseye

### DISCUSSION THREAD

**Discussion by mightyoakbob** [MusicPlayerDaemon/MPD#1432] [Q&A]
Hi Guys,

Apologies, I'm very new to this and know very little.

I'm using mpd on a Raspberry Pi and although it works lovely when built on Buster OS, I can't get the config right on Bullseye and I need advice. I'm using a hifiberry £30 dac and Pete Manchester's Mediaplayer.
https://github.com/PeteManchester/MediaPlayer

Following the same procedure on Bullseye as Buster doesn't work. 
The first sign of trouble is when you boot up the pie, the green (playing) led on the dac doesn't extinguish when the boot finishes and it remains on.

The iPad seems to be able to send music to the player or it tells me it does but it doesn't play.

On Buster I make very few changes to the default setting in mpd.conf and those are made following some instructions found on the net.

1. Uncomment 'group' and change group name: to: "audio”

2. Change bind_to_address		"localhost" to: "any"

With just those changes it will work fine but to get the volume control to work from the iPad controller I need to change one more line...

#       mixer_type      "hardware"
to:
        mixer_type      "software"

Those are all of the changes needed on Buster, it then fully works. 

With Bullseye, I had a look at the logs and it seems to be trying to use the jack output and failing to do so. Then I looked at the ALSA settings in mpd.conf and found that everything is REM'd out unlike in Buster. I tried changing the first ALSA settings to the same as in Buster but no joy and I returned them to default.

Lots and lots of google followed and I found a post here..
https://www.hydrus.org.uk/journal/raspberry-pi-11.html
Which details making a file called ".asoundrd" and placing it in 
/var/lib/mpd.

That turned out the green led and the system will now play but the volume control still doesn't work. So knowing that it relies on the mixer type being changed to software I tried to un-REM the first ALSA option again. Unfortunately, as soon as you un-rem anything there I'm back to square one with the green light stuck on.

If someone can help me, I would be very grateful.

Thanks.

Bob.

**Comment by MaxKellermann** [comment]
https://www.musicpd.org/help/

**Comment by mightyoakbob** [comment]
Hi, 
I'm sorry, obviously I've got it wrong. I've been to the link given and read it and fail to understand it. To me it doesn't make clear if I should post files one at a time or all in one go, it doesn't make clear how to link that back to my post here. It's obviously not inteneded for newbies, sorry to have bothered you.

Bob.

**Reply by MaxKellermann** [reply]
You asked for help. The link I gave you said "When you do request help, post the following data:" but your request for help contained none of that.
It is impossible to help you without any information about your setup and without a proper description of the problem.
You are only qualified for getting help if you can read and understand a sentence like "when you do request help, post the following data". I can't imagine somebody who can't understand that sentence, yet here you are.

**Reply by mightyoakbob** [reply]
Max, you are clearly familiar with how this site works, I'm very much not familiar and truefully I'm out of my depth.
I'm more than happy to post anything if I know what to do but I don't.
The message you gave me told me to post to "paste.debian.net". I don't understand how that connects back to here.
I have been to paste.debian.net and cannot work out if I send one file at a time or can somehow send multiple files.

Yes, more than willing to do as asked but apparently too thick, I can only appologise.

Bob.

**Reply by MaxKellermann** [reply]
> The message you gave me told me to post to "paste.debian.net".

No, it does not tell you that. This is not IRC, therefore this part does not apply to you.
You still havn't posted any information, and I'm tired of this off-topic discussion. Good luck.

**Comment by mightyoakbob** [comment]
I have uploaded mpd.conf and the version info and the error I get from the log file.

I hope I've got that right.

Bob.



[mpd-conf.txt](https://github.com/MusicPlayerDaemon/MPD/files/7986433/mpd-conf.txt)
[mpd-log.txt](https://github.com/MusicPlayerDaemon/MPD/files/7986434/mpd-log.txt)
[mpd --version.txt](https://github.com/MusicPlayerDaemon/MPD/files/7986436/mpd.--version.txt)

**Reply by MaxKellermann** [reply]
**Sigh**.
https://www.musicpd.org/help/ says
> make sure you have the latest MPD version (0.23.5)

... and your `mpd --version` output says:
> Music Player Daemon 0.22.6 (0.22.6)

Enough of your ignorance. I won't try to help you anymore. Good luck and bye.

**Reply by mightyoakbob** [reply]
It's the version that gets installed with Bullseye, the latest raspi os.

**Reply by skidoo23** [reply]
Gentlemen, please.

@mightyoakbob MPD project resources are limited and the most precious resource is Max time. We ("the users") should make his work as comfortable and pleasant as possible. If project lead and contributor #1 asks for a favor - with good reason - this request should be granted.

Back to problem: your mpd.conf has no configured audio output. Does your Linux distribution use ALSA or Pulse audio? Does your kernel support your DAC? aplay -l output?

**Reply by mightyoakbob** [reply]
I came for help because I don't know much.
I had a procedure to build a working system based on Buster and it worked fine which was reapeatable.
I tried to do the exact same thing with bullseye and it doesn't work.
Not knowing much I assumed (wrongly) that a later version of mpd that came with the later OS version would be the latest.
I've learnt this morning it wasn't. I have spent the last few hours trying to work where to get the latest MPD from and how to install it. I still have not succeeded but I've broken my system trying.

I cannot answer your questions other than to say it worked fine with the same hifiberry dac with Buster OS. I can get it to play with Bullseye using the little fix I mentioned in my first post. If using buster a tweak to alsa config makes the volume control work I assume its ALSA. 

I assume aplay -I output is a command to get info?
I will do that as soon as I can but I'm on my 3rd complete rebuild now trying to work out how the blazes to update mpd so the thing is broken at the moment.

I'm sorry I've wasted people's time but I don't know what I don't know and everyone is ignorant of something.

Thanks for trying to help, I'll run the command if I ever get the build to work again and I figure where the latest version is and how to insatll it.

Bob.

**Reply by kingosticks** [reply]
Raspberry Pi OS used to be ALSA only, now it ships with Pulseaudio (except the Lite version which still uses ALSA - but I am sure you would have mentioned if you were using the Lite version). To use Pulseaudio you need to jump through the hoops that allow an mpd service to connect to a Pulseaudio user service. Fortunately there's a very similar problem [here](https://forums.raspberrypi.com/viewtopic.php?p=1947206&sid=10e6fd2755ab62dadc461aca87cb6a6e#p1947206) and the same solution should work for you.

In general, since you are using an old version of mpd that came with Raspberry Pi OS, you should try and get support from the Raspberry Pi forums. Old versions are not supported here.

**Reply by mightyoakbob** [reply]
I'm sorry, I'm sorry, I'm sorry. I'm using the lite version. I had no idea that was important or even relevent.

I am trying very hard and failing to update to the latest version but failing because to a newbie it's very uphill.

Thanks.

Bob.

**Reply by kingosticks** [reply]
Don't bother updating, it won't fix your lack of configuration. Do what @skidoo23 already said: run the command `aplay -l` to see the index/name of your audio hardware device and then configure that device in an audio output section in mpd.conf

**Reply by mightyoakbob** [reply]
I have updated mpd, I think. The only place on the net that even hints how to do it seems out of date so I'm not sure at all. The green dac light is still stuck on so no improvement.

Here is the result of the command you wanted running.
aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: vc4hdmi [vc4-hdmi], device 0: MAI PCM i2s-hifi-0 [MAI PCM i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: sndrpihifiberry [snd_rpi_hifiberry_dacplus], device 0: HiFiBerry DAC+ Hi                                          Fi pcm512x-hifi-0 [HiFiBerry DAC+ HiFi pcm512x-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

This matches with the code that I mentioned in my first post that was changing the audio out from 0 to 1.
The problem though is that I need the ALSA setting of mixer_type "software" so that the volume control works.
I've no idea how to configure this in mpd.conf

I'd best go away hadn't I?

Bob.

**Comment by leigh123linux** [comment]
@mightyoakbob  You can use this command to find the correct device.

```
$ aplay --list-pcms |grep iec958
```



Example:

```
$ aplay --list-pcms |grep iec958
iec958:CARD=D10,DEV=0
```

```
audio_output {
	type             "alsa"
	name             "D10"
	device           "iec958:CARD=D10,DEV=0"
	auto_resample    "no"
	auto_format      "no"
	auto_channels    "no"
    replay_gain_handler "none"
    mixer_type       "none"
}
```

**Reply by mightyoakbob** [reply]
Leigh, Thanks so much, I'm sure your post is very close to what I need but again I'm too thick.
If I type: 
aplay --list-pcms |grep iec958
I get nothing back at all - nothing. It just comes back to $
So I thought perhaps the iec958 is specific to another device I don't have?
I just don't understand, sorry.

But thanks for trying.

Bob.

**Comment by leigh123linux** [comment]
Try it without grep

```
aplay --list-pcms
```

**Reply by mightyoakbob** [reply]
Right thanks Leigh, that does give an output...

 aplay --list-pcms
null
    Discard all samples (playback) or generate zero samples (capture)
sysdefault
    Default Audio Device
default
hw:CARD=vc4hdmi,DEV=0
    vc4-hdmi, MAI PCM i2s-hifi-0
    Direct hardware device without any conversions
plughw:CARD=vc4hdmi,DEV=0
    vc4-hdmi, MAI PCM i2s-hifi-0
    Hardware device with all software conversions
sysdefault:CARD=vc4hdmi
    vc4-hdmi, MAI PCM i2s-hifi-0
    Default Audio Device
hdmi:CARD=vc4hdmi,DEV=0
    vc4-hdmi, MAI PCM i2s-hifi-0
    HDMI Audio Output
dmix:CARD=vc4hdmi,DEV=0
    vc4-hdmi, MAI PCM i2s-hifi-0
    Direct sample mixing device
hw:CARD=sndrpihifiberry,DEV=0
    snd_rpi_hifiberry_dacplus, HiFiBerry DAC+ HiFi pcm512x-hifi-0
    Direct hardware device without any conversions
plughw:CARD=sndrpihifiberry,DEV=0
    snd_rpi_hifiberry_dacplus, HiFiBerry DAC+ HiFi pcm512x-hifi-0
    Hardware device with all software conversions
sysdefault:CARD=sndrpihifiberry
    snd_rpi_hifiberry_dacplus, HiFiBerry DAC+ HiFi pcm512x-hifi-0
    Default Audio Device
dmix:CARD=sndrpihifiberry,DEV=0
    snd_rpi_hifiberry_dacplus, HiFiBerry DAC+ HiFi pcm512x-hifi-0
    Direct sample mixing device

So what is the first part of the device string?
Am I getting there with the rest?

audio_output {
	type             "alsa"
	name             "sndrpihifiberry"
	device           "??????:CARD=sndrpihifiberry,DEV=0"
	auto_resample    "no"
	auto_format      "no"
	auto_channels    "no"
    replay_gain_handler "none"
    mixer_type       "none"
}

Thanks

Bob.

**Comment by leigh123linux** [comment]
I would try

```
hw:CARD=sndrpihifiberry,DEV=0
```

**Reply by mightyoakbob** [reply]
Unfortunately no change. DAC led still stuck on and no sound.
I've been trying to fix this for nearly 2 weeks now. So easy with Buster and near impssible with Bullseye. I realise that it's not the OS but other pckages linked to it but boy what %^%$ this is.

I've also noticed that hi-fiberry instructions say to edit the file:
/boot/config.txt
and to rem the line:
dtparam=audio=on
and add the line:
dtoverlay=hifiberry-dacplus

They say this will deactivate the default card o and that the berry dac will then becaone card 0. This is true, it does on Buster but not on Bullseye, card zero still there and berry dac is card 1.

Maybe if I could find a way to kill card 0 it may work like Buster.

Thanks Leigh.

Bob.

**Comment by leigh123linux** [comment]
Try removing the following options
```
auto_resample "no"
auto_format "no"
auto_channels "no"
```

**Reply by mightyoakbob** [reply]
Hi Leigh,
Where are they found? Not in my mpd.conf

Thanks 
Bob.

**Comment by Emilot** [comment]
@mightyoakbob give us your /boot/config.txt and type this at your command prompt:

aplay -l

and post the results please.

Thank you

**Reply by mightyoakbob** [reply]
Thank you. here it is.

aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: vc4hdmi [vc4-hdmi], device 0: MAI PCM i2s-hifi-0 [MAI PCM i2s-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: sndrpihifiberry [snd_rpi_hifiberry_dacplus], device 0: HiFiBerry DAC+ HiFi pcm512x-hifi-0 [HiFiBerry DAC+ HiFi pcm512x-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0

**Comment by Emilot** [comment]
Put these to your mpd.conf file..and try it...

mixer_type      "software"

audio_output {
enabled         "yes"
type            "alsa"
name            "MY DAC"
device          "hw:1,0"
auto_resample   "no"
auto_channels   "no"
auto_format     "no"
}

**Reply by mightyoakbob** [reply]
Thank you very much for that but unfortunately no change.

I'll pick this up in the morning, I have to go now, thanks to everyone.

Cheers,

Bob.

**Reply by mightyoakbob** [reply]
I've found the problem!
I'm sorry but it turned out to be nothing to do with mpd.
Unlike Buster, Bullseye enables another sound card in /boot/config.txt
That displaced the HiFiBerry dac to card 1 not card 0.
For some reason, all attempts to get the dac to work as card 1 failed.
The fix is to remove the line:
dtoverlay=vc4-kms-v3d

Sorry to have troubled you all and many thanks to those kind people who offered help and suggestions.

Bob.
